apps/src/templates/publishDialog/PublishDialog.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import Radium from 'radium';
+import Dialog, { Body, styles as dialogStyles } from '../Dialog';
+import PendingButton from '../PendingButton';
+import LegacyButton, { BUTTON_TYPES } from '../LegacyButton';
+import i18n from '@cdo/locale';
+import { hidePublishDialog, publishProject } from './publishDialogRedux';
+
+class PublishDialog extends Component {
+  static propTypes = {
+    isOpen: PropTypes.bool.isRequired,
+    isPublishPending: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    onConfirmPublish: PropTypes.func.isRequired,
+    projectId: PropTypes.string,
+    projectType: PropTypes.string,
+  }
+
+  confirm = () => {
+    this.props.onConfirmPublish(
+      this.props.projectId,
+      this.props.projectType,
+    );
+  };
+
+  close = () => this.props.onClose();
+
+  render() {
+    return (
+      <Dialog
+        title={i18n.publishToPublicGallery()}
+        isOpen={this.props.isOpen}
+        handleClose={this.close}
+      >
+        <Body>
+          <div style={[dialogStyles.body, {marginBottom: 10}]}>
+            {i18n.publishToPublicGalleryWarning()}
+          </div>
+          <LegacyButton type="cancel" onClick={this.close}>
+            {i18n.dialogCancel()}
+          </LegacyButton>
+          <PendingButton
+            isPending={this.props.isPublishPending}
+            onClick={this.confirm}
+            pendingStyle={BUTTON_TYPES.default.style}
+            pendingText={i18n.publishPending()}
+            style={[BUTTON_TYPES.primary.style, {float: 'right'}]}
+            text={i18n.publish()}
+          />
+        </Body>
+      </Dialog>
+    );
+  }
+}
+export default connect(state => ({
+  isOpen: state.publishDialog.isOpen,
+  isPublishPending: state.publishDialog.isPublishPending,
+  projectId: state.publishDialog.projectId,
+  projectType: state.publishDialog.projectType,
+}), (dispatch, ownProps) => ({
+  onClose() {
+    dispatch(hidePublishDialog());
+  },
+  onConfirmPublish(projectId, projectType) {
+    dispatch(publishProject(projectId, projectType));
+  },
+}), (stateProps, dispatchProps, ownProps) => (
+  // Allow stories and tests to override connected props.
+  Object.assign({}, stateProps, dispatchProps, ownProps)
```

Comment:
"Simply moving this logic into the component isn't an option because it won't have access to ownProps there. mergeProps to the rescue!"

I'm not sure I understand this. Why would the component not have access to ownProps? `ownProps` is just the set of props that get passed into the component directly (instead of through redux), right?

Also not sure I fully grok why this mergeProps is necessary here. Our strategy elsewhere has been to export the unconnected version of the component (used in tests/stories) and default export the connected version.